superset/db_engine_specs.py

```diff
@@ -639,7 +641,28 @@ def extra_table_metadata(cls, database, table_name, schema_name):
         }
 
     @classmethod
-    def handle_cursor(cls, cursor, query, session):
+    def prefetch_results(cls, cursor, query, cache_timeout, session, limit=1000):
+        data = cursor.fetchmany(limit)
+        cdf = utils.convert_results_to_df(cursor.description, data)
+        payload = dict(query_id=query.id)
+        payload.update({
+            'status': utils.QueryStatus.PREFETCHED,
+            'data': cdf.data if cdf.data else [],
```

Comment:
I don't think you need the if/else as `cdf.data` returns an empty array if the data frame is empty per [this](https://github.com/apache/incubator-superset/blob/master/superset/dataframe.py#L55). Also `x or []` is preferred over `x if x else []`.